### PR TITLE
Defer all possible requirements tests to super.canPlay()

### LINKED ISSUE #3234: [Agendas] Scientists P4 sometimes not reducing science tag requirements.
"Turmoil game with custom policy: Scientists policy(#4) is not working properly. The science tag reduction policy is not reducing the science tag requirements." Seems to be something that only happens infreqently

### PR DESCRIPTION
**Pull request by kberg**
Fixes #3234

### CODE REVIEW

**Comment by kberg** [comment]
I also restructred many of the canPlay checks that deal with reds party costs for readability.

File: src/cards/base/ArtificialLake.ts [hunk 1]
```diff
@@ -31,15 +31,18 @@ export class ArtificialLake extends Card implements IProjectCard {
     });
   }
   public canPlay(player: Player): boolean {
-    const meetsRequirements = super.canPlay(player);
+    if (!super.canPlay(player)) {
```
Comment (bafolts) [inline src/cards/base/ArtificialLake.ts:34]: Awesome. This is also a performance improvement as well.

File: src/cards/venusNext/SpinInducingAsteroid.ts [hunk 2]
```diff
@@ -30,13 +30,15 @@ export class SpinInducingAsteroid extends Card implements IProjectCard {
...
 
   public canPlay(player: Player): boolean {
-    const meetsVenusRequirements = super.canPlay(player);
+    if (!super.canPlay(player)) {
+      return false;
+    }
 
     if (PartyHooks.shouldApplyPolicy(player.game, PartyName.REDS)) {
-      return player.canAfford(player.getCardCost(this) + REDS_RULING_POLICY_COST * 2, {titanium: true}) && meetsVenusRequirements;
+      return player.canAfford(player.getCardCost(this) + REDS_RULING_POLICY_COST * 2, {titanium: true}) && true;
```
Comment (bafolts) [inline src/cards/venusNext/SpinInducingAsteroid.ts:38]: Is this `&& true` necessary?

**Review by bafolts — APPROVED** [review]

File: src/cards/venusNext/SpinInducingAsteroid.ts (hunk 2 above)
Comment (kberg) [reply src/cards/venusNext/SpinInducingAsteroid.ts:38]: No, good catch.

**Pull request merged**